### PR TITLE
6529151: NullPointerException in swing.plaf.synth.SynthLookAndFeel$Handler

### PR DESCRIPTION
**Pull request by prsadhuk**
NPE is thrown for closed/disposed JDialogs after uninstalling the UI which is because for SynthEditorPaneUI even though is disposed of, Nimbus propertyHandler tries to get the style of this disposed component resulting in NPE, 
which is fixed by a null check to prevent deferencing style object to get the color.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-6529151](https://bugs.openjdk.org/browse/JDK-6529151): NullPointerException in swing.plaf.synth.SynthLookAndFeel$Handler

### Reviewers
 * [Sergey Bylokhov](https://openjdk.org/census#serb) (@mrserb - **Reviewer**) ⚠️ Review applies to [f62d58b9](https://git.openjdk.org/jdk/pull/10167/files/f62d58b966f2c27985b2625295e1d397fd0cb62a)
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Author) ⚠️ Review applies to [f62d58b9](https://git.openjdk.org/jdk/pull/10167/files/f62d58b966f2c27985b2625295e1d397fd0cb62a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10167/head:pull/10167` \
`$ git checkout pull/10167`

Update a local copy of the PR: \
`$ git checkout pull/10167` \
`$ git pull https://git.openjdk.org/jdk pull/10167/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10167`

View PR using the GUI difftool: \
`$ git pr show -t 10167`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10167.diff">https://git.openjdk.org/jdk/pull/10167.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/swing/plaf/synth/RepaintNPE.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+            Component[] components = container.getComponents ();
+
+            for ( Component comp : components ) {
+                // Special dispose for JComponent's remove UI
+                if (comp instanceof JComponent) {
+                    JComponent jComp = (JComponent)comp;
+
+                    try
+                    {
+                        Method method = jComp.getClass().getMethod ( "getUI" );
```
Comment (mrserb) [inline test/jdk/javax/swing/plaf/synth/RepaintNPE.java:84]: The getUI() method is public, can be used directly.

Comment (prsadhuk) [reply test/jdk/javax/swing/plaf/synth/RepaintNPE.java]: Yes. modified.

**Review by mrserb — APPROVED** [review]

**Review by DamonGuy — APPROVED** [review]
Applied changes and NPE is no longer. Looks good to me

**Pull request closed without merging**